Scala/syntax_test_scala.scala

```diff
@@ -1329,12 +1329,30 @@ trait AlgebraF[F[_]] { type f[x] = Algebra[F,x] }
 //                     ^^^^ storage.type.scala
 //                               ^ keyword.operator.assignment.scala
 
+class Foo @Inject()(a: String)
+//         ^^^^^^ meta.annotation.identifier
+//                 ^- meta.annotation
+//                  ^ variable.parameter.scala
+
+class Foo @Inject() @Provided(scope = "test")(a: String)
+//         ^^^^^^ meta.annotation.identifier
+//                 ^- meta.annotation
+//                   ^^^^^^^ meta.annotation.identifier
+//                           ^^^^^^^ meta.annotation.parameters
+//                                           ^- meta.annotation
+//                                            ^ variable.parameter.scala
+
+class Foo @Inject(z = "a")(a: String)
+//         ^^^^^^ variable.annotation
+//                ^^^^^^^ meta.annotation.parameters
+//                         ^ variable.parameter.scala
+
```

Comment:
Can you also try the following?

```scala
class Foo @Inject() [A](a: String)
```

Just to ensure we stay at the appropriate point in the state machine. Or are these only valid *after* type parameters? I've honestly never used this syntax.

Reply:
This code is invalid. Did you mean:

```scala
class Foo[A] @Inject()(a: String)
```

Reply:
Nice! Like I said, I never use this syntax so I don't know what is and is not valid. :-) But yes, that would be a good one to add, basically just covering the last case in the state machine.